lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -4641,6 +4641,104 @@ LogicalResult AtenNormScalarOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// AtenRenormOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult AtenRenormOp::verify() {
+
+  auto selfType = cast<BaseTensorType>(getSelf().getType());
+
+  if (!selfType.hasDtype() || !selfType.hasSizes()) {
+    return success();
+  }
+
+  auto inShape = selfType.getSizes();
+  int64_t selfRank = inShape.size();
+  auto selfDtype = selfType.getDtype();
+
+  // Check if dtype is one of those supported by renorm operation.
+  // ComplexType will match any torch complex types, but each float must be
+  // checked individually.
+  if (selfDtype.isa<mlir::ComplexType>()) {
+    return emitOpError(
```

Comment:
If you include this verification here, do you also need to check for the corresponding match failure in the decompose complex ops pattern?

Reply:
Or rather, since this is just something which is unsupported in the lowering, but not for the original op, should it be included in the verifier after all?

Reply:
You're right. This is unsupported in lowering so it needs to be checked in Decompose only.